pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -238,135 +239,846 @@ func TestReconcile(t *testing.T) {
+					MountPath: "/builder/home",
+				},
+				},
+			}},
+		}, {
+			name:                   "taskrun-with-resource-spec-task-spec",
+			taskRun:                taskRunWithResourceSpecAndTaskSpec,
+			wantServiceAccountName: "",
+			wantPodVolume:          getVolumes(),
+			wantSteps: []corev1.Container{{
+				Name:       "build-step-credential-initializer-mz4c7",
+				Image:      "override-with-creds:latest",
+				Args:       []string{},
+				WorkingDir: "/workspace",
+				Env: []corev1.EnvVar{{
+					Name:  "HOME",
+					Value: "/builder/home",
+				},
+				},
+				VolumeMounts: []corev1.VolumeMount{{
+					Name:      "workspace",
+					ReadOnly:  false,
+					MountPath: "/workspace",
+				}, {
+					Name:      "home",
+					ReadOnly:  false,
+					MountPath: "/builder/home",
+				},
+				},
+			}, {
+				Name:       "build-step-git-source-workspace-9l9zj",
+				Image:      "override-with-git:latest",
+				Args:       []string{"-url", "github.com/build-pipeline.git", "-revision", "rel-can", "-path", "/workspace/workspace"},
+				WorkingDir: "/workspace",
+				Env: []corev1.EnvVar{{
+					Name:      "HOME",
+					Value:     "/builder/home",
+					ValueFrom: (*corev1.EnvVarSource)(nil),
+				},
+				},
+				VolumeMounts: []corev1.VolumeMount{{
+					Name:      "workspace",
+					ReadOnly:  false,
+					MountPath: "/workspace",
+				}, {
+					Name:      "home",
+					ReadOnly:  false,
+					MountPath: "/builder/home",
+				},
+				},
+			}, {
+				Name:       "build-step-place-tools",
+				Image:      "gcr.io/k8s-prow/entrypoint@sha256:7c7cd8906ce4982ffee326218e9fc75da2d4896d53cabc9833b9cc8d2d6b2b8f",
+				Command:    []string{"/bin/cp"},
+				Args:       []string{"/entrypoint", "/tools/entrypoint"},
+				WorkingDir: "/workspace",
+				Env: []corev1.EnvVar{
+					{
+						Name:  "HOME",
+						Value: "/builder/home",
+					},
+				},
+				VolumeMounts: []corev1.VolumeMount{{
+					Name:      "tools",
+					ReadOnly:  false,
+					MountPath: "/tools",
+				}, {
+					Name:      "workspace",
+					ReadOnly:  false,
+					MountPath: "/workspace",
+				}, {
+					Name:      "home",
+					ReadOnly:  false,
+					MountPath: "/builder/home",
+				},
+				},
+			}, {
+				Name:       "build-step-mystep",
+				Image:      "ubuntu",
+				Args:       []string{},
+				Command:    []string{"/tools/entrypoint"},
+				WorkingDir: "/workspace",
+				Env: []corev1.EnvVar{
+					{
+						Name:  "HOME",
+						Value: "/builder/home",
+					}, {
+						Name:  "ENTRYPOINT_OPTIONS",
+						Value: "{\"args\":[\"mycmd\"],\"process_log\":\"/tools/process-log.txt\",\"marker_file\":\"/tools/marker-file.txt\"}",
+					},
+				},
+				VolumeMounts: []corev1.VolumeMount{{
+					Name:      "tools",
+					ReadOnly:  false,
+					MountPath: "/tools",
+				}, {
+					Name:      "workspace",
+					ReadOnly:  false,
+					MountPath: "/workspace",
+				}, {
+					Name:      "home",
+					ReadOnly:  false,
+					MountPath: "/builder/home",
+				},
+				},
+			}},
+		},
+	} {
```

Comment:
i think if we _wanted_ to we could make some of these a bit less verbose but this is fine for now if we want to iterate on it :D

Reply:
yes, agreed, just wanted to get everything working first, then we can improve on it